File: 0205/solution.ts
function isIsomorphic(s: string, t: string): boolean {
  if (s.length !== t.length) {
    return false;
  }

  const sMap: Map<string, string> = new Map();
  const tMap: Map<string, string> = new Map();
  for (let i = 0; i < s.length; i++) {
    const sChar = s[i];
    const tChar = t[i];
    if (sMap.has(sChar) && sMap.get(sChar) !== tChar) {
      return false;
    } else {
      sMap.set(sChar, tChar);
    }
    if (tMap.has(tChar) && tMap.get(tChar) !== sChar) {
      return false;
    } else {
      tMap.set(tChar, sChar);
    }
  }
  return true;
}

File: 0205/solution.go
package main

func isIsomorphic(s string, t string) bool {
	sMap := make(map[byte]byte)
	tMap := make(map[byte]byte)
	if len(s) != len(t) {
		return false
	}

	for i := 0; i < len(s); i++ {
		val, ok := sMap[s[i]]
		if ok && val != t[i] {
			return false
		} else {
			sMap[s[i]] = t[i]
		}
		val, ok = tMap[t[i]]
		if ok && val != s[i] {
			return false
		} else {
			tMap[t[i]] = s[i]
		}
	}
	return true
}